core/src/serde_support.rs

```diff
@@ -6,20 +6,23 @@
 
 use alloc::format;
 use alloc::string::String;
+use alloc::vec::Vec;
 
-use base64::engine::general_purpose::STANDARD as BASE64_STANDARD;
-use base64::Engine;
-use serde::de::{Error as SerdeError, MapAccess, Unexpected, Visitor};
+use phoenix_core::Note;
+use piecrust_uplink::ContractId;
```

Comment:
You can access ContractId within this crate, instead of referring to `piecrust_uplink`

```suggestion
use crate::abi::ContractId;
```